ecpp/license_POST.ecpp

```diff
@@ -142,7 +144,45 @@ bool database_ready;
             log_error(ret.error().c_str());
             http_die("internal-error", ret.error().c_str());
         }
-        auto rv_svc = proc.wait(15000);
+
+        // The start-db-services script ensures that all FTY/IPM2 related
+        // services and targets become active, including that the database
+        // schema gets initialized and its chain of consumers starts,
+        // and we are blocked until we "know" there is now someone running
+        // that would actually handle the backend requests for the rest of
+        // EULA wizard.
+        // So as far as we are concerned here, the script runtime can last
+        // several minutes (even if the CPU/IO/Time load is mostly elsewhere).
+        // Normally tntnet would suicide with a 10-minute request processing,
+        // unless a deadman timer is tickled to confirm we are stoll alive.
+        // This can be tuned with maxRequestTime (0 to disable the watchdog)
+        // or handled safely with a loop sleeping, checking that the script
+        // still runs, and tickling tntnet::Worker::touch() in the known-long
+        // processing loop. It should suffice to catch `unexpected("timeout")`
+        // (child still running as of last check before time limit; no other
+        // errors) as per fty-utils::fty/process.h wording of Process::wait().
+        // the wait() would end earlier if the child process finishes:
+        int wait_cycle = 3 * 60000;
+
+        // configured in seconds
+        unsigned int mrt = tnt::TntConfig::it().maxRequestTime;
+        if (mrt > 0 && mrt < (INT_MAX / 1000)) { // if enabled at all
+            wait_cycle = static_cast<int>(mrt) * 666 ; // convert to msec and trim 2/3 for overheads
```

Comment:
SAtan is watching you?

Reply:
Sure, 1000 * 2/3 in integer maths, to allow decent overhead between timing this and context switches and other activities :)

Reply:
In later revisions, made this behavior optional. It turned out to be better to use shorter sleep cycles (to not allow blocking for long on child process `write()`s, and drain their buffers quickly and regularly).